### PR TITLE
Fixes #820: maintains project.archived value when their org is archived/unarchived

### LINKED ISSUE #820: Already archived projects get unarchived when org is unarchived
### Steps to reproduce the error
1. In an active org with an active project, archive the project.
2. Archive the org.
3. Unarchive the org.
### Actual behavior

The project archived in step 1 gets unarchived.
### Expected behavior

The project archived in step 1 should remain archived. Since this project was archived before the org was archived, it should remain archived irrespective of the org's archived state unless it was specifically unarchived.
### Possible solution

Don't archive the projects when an org is archived. But treat projects in an archived org as if they were archived as well regardless of the project's actual archived state.

### PR DESCRIPTION
**Pull request by laura-barluzzi**
### Proposed changes in this pull request

This PR fixes #820 _Already archived projects get unarchived when org is unarchived_ by:
- Removing the lines of code that manually set all `project.archived=True` when archiving an organization in `organization/views/default.OrgArchiveView.archive()`
- Adding `organization__archived=True` as an additional filter value for public projects in `organization/views/default.ProjectList.get()` so that all projects of archived organizations aren't displayed, despite their own `.archived` field setting. 
- Changing the query in `if is_admin:` block using the _OR query logic_ with [Q objects](https://docs.djangoproject.com/en/dev/topics/db/queries/#complex-lookups-with-q-objects). 
```
filter(Q(archived=True) | Q(organization__archived=True))
```
This simply adds the _archived projects of not-archived organizations_ and _all projects of archived organizations_.
- Adding the `Archived` red flag label next to the `organization.name` in both `organization/project_wrapper.html` and `organization/project_list.html` templates to better visualize the situation. This is mainly useful in case there are not-archived projects in an archived organization -- the missing project `Archived` red flag label is substituted by the one for their organization.
- Changing 2 assert statements in `OrganizationUnarchiveTest` and `OrganizationArchiveTest` to meet the new logic for which the archiving of projects doesn't take place.

### When should this PR be merged
- When it is reviewed and approved.

### Risks
- Low

### Follow-up actions
- Double check other locations where the same logic could be applied -- e.g. are we showing archived organization projects to users who don't have permission for that somewhere else?
- Close issue Already archived projects get unarchived when org is unarchived #820

#### General

- **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description. 
  - [ ] Review 1
  - [ ] Review 2
- **Is the PR labeled correctly?** It should have the `migration` label if a new migration is added. 
  - [ ] Review 1
  - [ ] Review 2
- **Is the risk level assessment sufficient?** The risks section should contain all risks that might be introduced with the PR and which actions we need to take to mitigate these risks. Possible risks are database migrations, new libraries that need to be installed or changes to deployment scripts.
  - [ ] Review 1
  - [ ] Review 2

#### Functionality

- **Are all requirements met?** Compare implemented functionality with the requirements specification.
  - [ ] Review 1
  - [ ] Review 2
- **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled. 
  - [ ] Review 1
  - [ ] Review 2

#### Code

- **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
  - [ ] Review 1
  - [ ] Review 2
- **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests. 
  - [ ] Review 1
  - [ ] Review 2
- **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/). 
  - [ ] Review 1
  - [ ] Review 2
- **Is the code documented sufficiently?** Large and complex classes, functions or methods must be annotated with comments following our [code-style guidelines](https://devwiki.corp.cadasta.org/Contributing%20Style%20Guide#documentation-and-comments).
  - [ ] Review 1
  - [ ] Review 2
- **Has the scalability of this change been evaluated?**
  - [ ] Review 1
  - [ ] Review 2
- **Is there a maintenance plan in place?**
  - [ ] Review 1
  - [ ] Review 2

#### Tests

- **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components. 
  - [ ] Review 1
  - [ ] Review 2
- **If this is a bug fix, are tests for the issue in place?**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
  - [ ] Review 1
  - [ ] Review 2
- **If this is a new feature or a significant change to an existing feature?** has the manual testing spreadsheet been updated with instructions for manual testing?
  - [ ] Review 1
  - [ ] Review 2

#### Security

- **Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
  - [ ] Review 1
  - [ ] Review 2
- **Are all UI and API inputs run through forms or serializers?** 
  - [ ] Review 1
  - [ ] Review 2
- **Are all external inputs validated and sanitized appropriately?**
  - [ ] Review 1
  - [ ] Review 2
- **Does all branching logic have a default case?**
  - [ ] Review 1
  - [ ] Review 2
- **Does this solution handle outliers and edge cases gracefully?**
  - [ ] Review 1
  - [ ] Review 2
- **Are all external communications secured and restricted to SSL?**
  - [ ] Review 1
  - [ ] Review 2

#### Documentation

- **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes must be documented in the [Cadasta Platform Documentation](https://github.com/Cadasta/cadasta-docs).
  - [ ] Review 1
  - [ ] Review 2
- **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented in the [API docs](https://github.com/Cadasta/api-docs).
  - [ ] Review 1
  - [ ] Review 2
- **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki. 
  - [ ] Review 1
  - [ ] Review 2

### CODE REVIEW

File: cadasta/templates/organization/project_list.html [hunk 1]
```diff
@@ -70,7 +70,10 @@
             <img src="{{ proj.organization.logo }}" class="org-logo" alt="{{ proj.organization.name }}"/>
           </div>
         {% else %}
-          <p class="org-name-alt">{{ proj.organization.name }}</p>
+          <span class="org-name-alt">{{ proj.organization.name }}</span>
+          {% if proj.organization.archived %}
```
Comment (seav) [inline cadasta/templates/organization/project_list.html]: If the organization has a logo, this archived label does not appear. (Admittedly, we can't upload org logos now, but this should still be considered.)

Comment (laura-barluzzi) [inline cadasta/templates/organization/project_list.html]: Sure thing. I didn't notice. For now I move the label outside, so it appears also with logos :+1:

**Comment by laura-barluzzi** [comment]
@amplifi thanks for your review. I went down this road mainly because the #820 issue has been opened for 10 months and its description states: 

> Possible solution
> Don't archive the projects when an org is archived. But treat projects in an archived org as if they were archived as well regardless of the project's actual archived state.

No comments were added in the thread, so I believed this was the desired approach to apply. This said,  I have in mind another potential solution:
- add `Project.org_archived` field: this field is by default False, and if the org is archived we set it to True (this happens where currently we set to True the `proj.archived`)
- whenever we display throughout the platform the project Archived flag, we use this if statement {% if proj.archived or proj.org_archived %} instead of checking only `proj.archived`

I try to change it today, it should be fairly quick and easy to do.

**Comment by laura-barluzzi** [comment]
I close this and open a new one.

**Pull request closed without merging**